source/extensions/filters/network/rbac/rbac_filter.cc

```diff
@@ -83,8 +85,33 @@ void RoleBasedAccessControlFilter::setDynamicMetadata(std::string shadow_engine_
 EngineResult
 RoleBasedAccessControlFilter::checkEngine(Filters::Common::RBAC::EnforcementMode mode) {
   const auto engine = config_->engine(mode);
+
   if (engine != nullptr) {
     std::string effective_policy_id;
+    if (mode == Filters::Common::RBAC::EnforcementMode::Enforced) {
+      // Check log decision
+      LogDecision log_dec = engine->shouldLog(
+          callbacks_->connection(), callbacks_->connection().streamInfo(), &effective_policy_id);
+      if (log_dec != LogDecision::Undecided) {
+        ProtobufWkt::Struct log_metadata;
+        auto& fields = *log_metadata.mutable_fields();
+        fields[Filters::Common::RBAC::DynamicMetadataKeysSingleton::get().AccessLogKey]
+            .set_bool_value(log_dec == LogDecision::Yes);
+        callbacks_->connection().streamInfo().setDynamicMetadata(
+            Filters::Common::RBAC::DynamicMetadataKeysSingleton::get().CommonNamespace,
+            log_metadata);
+
+        if (log_dec == LogDecision::Yes) {
+          ENVOY_LOG(debug, "request logged");
+          config_->stats().logged_.inc();
+        } else {
+          ENVOY_LOG(debug, "request not logged");
+          config_->stats().not_logged_.inc();
+        }
+      }
+    }
```

Comment:
This is redundant between filters. Can you unify into a base class / helper?

Reply:
I think currently there's alot of redundancy between network and http rbac filters. I'll consider how best to do this.